pdns/communicator.cc

```diff
@@ -92,6 +92,19 @@ void CommunicatorClass::go()
       exit(1);
     }
   }
+
```

Comment:
This is basically a copy of the for-loop right above, perhaps we can factor this out?